ims.yaml

```diff
@@ -0,0 +1,672 @@
+      type:
+        type: string
+        enum:
+          - object
+      meta:
+        $ref: '#/definitions/MetadataResource'
+      attributes:
+        type: object
+        properties:
+          status:
+            description: "All objects MUST BE either 'active' or tobedeleted. Something which is flagged 'tobedeleted' is to be considered safe to delete. Systems can delete records that are flagged as such if they wish, but they are not under any compulsion to do so."
+            type: string
+            enum:
+              - ACTIVE
+              - TOBEDELETED
+          dateLastModified:
+            description: 'Model Primitive Datatype = DateTime. All objects MUST be annotated with the dateTime upon which they were last modified. This enables requesters to query for just the latest objects. DateTimes MUST be expressed in W3C profile of [ISO 8601] and MUST contain the UTC timezone.'
+            type: string
+            format: date-time
+          formattedName:
+            description: ...tbd (ComplexType property description)...
+            type: array
+            minItems: 1
+            items:
+              $ref: '#/definitions/FormattedNameResource'
+          name:
+            description: ...tbd (ComplexType property description)...
+            type: array
+            minItems: 1
+            items:
+              $ref: '#/definitions/NameResource'
+          address:
+            description: ...tbd (ComplexType property description)...
+            type: array
+            minItems: 0
+            items:
+              $ref: '#/definitions/AddressResource'
+          contactInfo:
+            description: ...tbd (ComplexType property description)...
+            type: array
+            minItems: 0
+            items:
+              $ref: '#/definitions/ContactInfoResource'
+          demographics:
+            $ref: '#/definitions/DemographicsResource'
+          role:
+            description: ...tbd (ComplexType property description)...
+            type: array
+            minItems: 0
+            items:
+              $ref: '#/definitions/RoleResource'
+          agent:
+            description: ...tbd (ComplexType property description)...
+            type: array
+            minItems: 0
+            items:
+              $ref: '#/definitions/AgentResource'
+          credentials:
+            description: ...tbd (ComplexType property description)...
+            type: array
+            minItems: 0
+            items:
+              $ref: '#/definitions/CredentialsResource'
+        required:
+          - status
+          - dateLastModified
+          - formattedName
+          - name
```

Comment:
For the list of fields marked as required that we were given, I took that to mean "these are the fields that should be required when creating a new resource" so I used them in my post endpoint. For the response resource, I had all fields be required since we should never omit any of these fields in our api response. I talked with Miguel about this a little and he said one of the concerns was that clients shouldn't have to test if a response has a certain field before trying to access its value. Instead of omitting them, we could just set them to `null` or something. What are your thoughts?

Reply:
I agree that those should be interpreted as "required to post". If we should never omit fields in our api response, then we should probably remove the 'field's parameter which limits what is returned. Because this parameter exists, I made propose that none of the attributes are required for the response. My concern with setting them to null is that could be interpreted as "the resource doesn't have a value for that attribute" rather than "we did not return the value for that attribute". An alternative could be to remove the fields parameter and require everything in the response.

Reply:
Ah right, I completely forgot about the `fields` parameter, nice catch. I guess this corresponds to jsonapi's [sparse fieldsets](https://jsonapi.org/format/#fetching-sparse-fieldsets). I agree that we should make all attributes optional in the response if we're supporting this parameter. Since this functionality was in the original spec, and it corresponds to a jsonapi field, I think we should keep it for now.